airflow/sensors/hdfs_sensor.py

```diff
@@ -17,101 +17,231 @@
+class HdfsFileSensor(BaseSensorOperator):
+    """Waits for file(s) to land in HDFS."""
 
+    template_fields = ("_pattern",)
+    ui_color = settings.WEB_COLORS["LIGHTBLUE"]
+
+    @deprecated_args(
+        renamed={
+            "filepath": "file_pattern",
+            "hdfs_conn_id": "conn_id",
+            "file_size": "min_size",
+            "ignored_ext": "ignore_exts",
+        },
+        dropped={"ignore_copying", "hook"},
+    )
     @apply_defaults
-    def __init__(self,
-                 filepath,
-                 hdfs_conn_id='hdfs_default',
-                 ignored_ext=['_COPYING_'],
-                 ignore_copying=True,
-                 file_size=None,
-                 hook=HDFSHook,
-                 *args,
-                 **kwargs):
-        super(HdfsSensor, self).__init__(*args, **kwargs)
-        self.filepath = filepath
-        self.hdfs_conn_id = hdfs_conn_id
-        self.file_size = file_size
-        self.ignored_ext = ignored_ext
-        self.ignore_copying = ignore_copying
-        self.hook = hook
-
-    @staticmethod
-    def filter_for_filesize(result, size=None):
-        """
-        Will test the filepath result and test if its size is at least self.filesize
-
-        :param result: a list of dicts returned by Snakebite ls
-        :param size: the file size in MB a file should be at least to trigger True
-        :return: (bool) depending on the matching criteria
-        """
-        if size:
-            log = LoggingMixin().log
-            log.debug(
-                'Filtering for file size >= %s in files: %s',
-                size, map(lambda x: x['path'], result)
-            )
-            size *= settings.MEGABYTE
-            result = [x for x in result if x['length'] >= size]
-            log.debug('HdfsSensor.poke: after size filter result is %s', result)
-        return result
-
-    @staticmethod
-    def filter_for_ignored_ext(result, ignored_ext, ignore_copying):
-        """
-        Will filter if instructed to do so the result to remove matching criteria
-
-        :param result: (list) of dicts returned by Snakebite ls
-        :param ignored_ext: (list) of ignored extensions
-        :param ignore_copying: (bool) shall we ignore ?
-        :return: (list) of dicts which were not removed
-        """
-        if ignore_copying:
-            log = LoggingMixin().log
-            regex_builder = "^.*\.(%s$)$" % '$|'.join(ignored_ext)
-            ignored_extentions_regex = re.compile(regex_builder)
-            log.debug(
-                'Filtering result for ignored extensions: %s in files %s',
-                ignored_extentions_regex.pattern, map(lambda x: x['path'], result)
-            )
-            result = [x for x in result if not ignored_extentions_regex.match(x['path'])]
-            log.debug('HdfsSensor.poke: after ext filter result is %s', result)
-        return result
+    def __init__(
+        self,
+        pattern,
```

Comment:
_I added a few deprecation utility classes/functions to help keep the re-written classes as backwards compatible as possible._

This is not the case here:  `__init__(self, filepath, ..) -> __init__(self, pattern .. )`

Reply:
And this is not consistent with the `deprecation args`: `"filepath": "file_pattern",`